bin/pre-commit-hook.js

```diff
@@ -6,35 +6,75 @@ const execSync = require( 'child_process' ).execSync;
 const spawnSync = require( 'child_process' ).spawnSync;
 const chalk = require( 'chalk' );
 
-const gitFiles = execSync( 'git diff --cached --name-only --diff-filter=ACM' )
-	.toString()
-	.split( '\n' )
-	.map( name => name.trim() );
-
 /**
- * Parses the output of a git diff command into javascript file paths.
+ * Parses the output of a git diff command into file paths.
  *
  * @param   {String} command Command to run. Expects output like `git diff --name-only […]`
  * @returns {Array}          Paths output from git command
  */
 function parseGitDiffToPathArray( command ) {
 	return execSync( command, { encoding: 'utf8' } )
 		.split( '\n' )
-		.map( name => name.trim() )
-		.filter( name => name.startsWith( '_inc/client/' ) && /\.jsx?$/.test( name ) );
+		.map( name => name.trim() );
+}
+
+/**
+ * Provides filter to determine which PHP files to run through phpcs.
+ *
+ * @param {String} file File name of php file modified.
+ * @return {boolean}        If the file matches the whitelist.
+ */
+function phpcsFilesToFilter( file ) {
+	if (
```

Comment:
What would you say for having an array of whitelisted paths instead, like so:

```diff
diff --git a/bin/pre-commit-hook.js b/bin/pre-commit-hook.js
index 7f8e8b281..40bd3289c 100644
--- a/bin/pre-commit-hook.js
+++ b/bin/pre-commit-hook.js
@@ -25,10 +25,13 @@ function parseGitDiffToPathArray( command ) {
  * @return {boolean}        If the file matches the whitelist.
  */
 function phpcsFilesToFilter( file ) {
-	if (
-		file.startsWith( '_inc/lib/debugger/' )
-		// || file.startsWith( 'jetpack.php' ) // Example for future editions.
-	) {
+	// If the file path starts with anything like in the array below, it should be linted.
+	const whitelist = [
+		'_inc/lib/debugger/',
+		'extensions/',
+	];
+
+	if ( -1 !== whitelist.findIndex( filePath => file.startsWith( filePath ) ) ) {
 		return true;
 	}
 
```

Reply:
No strong opinion and your method looks cleaner. I'm not sure if there would be need for things other than `startsWith`, but I'm fine being opinionated there. I'll make the change.

Reply:
Added via e6cdcbd3c , including the extensions directory, but we should do a quick phpcs run on that to make it fully clean. There are a couple of what looks to be minor things at a glance.

Reply:
PHPCS issues to be resolved via #11383